src/jdk.jdwp.agent/share/native/libjdwp/eventHandler.c

```diff
@@ -1708,6 +1685,19 @@ eventHandler_reset(jbyte sessionID)
         }
     }
 
+    /* We also want to disable VIRTUAL_THREAD_START events if they were enabled due to
+     * a deferred event request.
+     */
+    if (gdata->virtualThreadStartEventsPermanentlyEnabled) {
```

Comment:
Looks like this block can disable VITRUAL_THREAD_START when `rememberVThreadsWhenDisconnected` is set.

Reply:
virtualThreadStartEventsPermanentlyEnabled is only set true if includeVThreads is false, which means rememberVThreadsWhenDisconnected is also false (they are always set the same, as I discuss in my new comment below).

virtualThreadStartEventsPermanentlyEnabled could use a better name because it means "permanently enabled for handling deferred event enabling", whereas if includeVThreads is true, VITRUAL_THREAD_START events are always permanently enabled, but virtualThreadStartEventsPermanentlyEnabled will never be set true.

Reply:
How about virtualThreadStartEventsEnabledForDeferredEventMode?

Reply:
> How about virtualThreadStartEventsEnabledForDeferredEventMode?

Agree with Alex, long but cleaner.